// mailersend/lib/modules/MailerSend.module.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.MailerSend = void 0;
const Email_module_1 = require("./Email.module");
const Token_module_1 = require("./Token.module");
const EmailVerification_module_1 = require("./EmailVerification.module");
const SMS_module_1 = require("./SMS.module");
const Others_module_1 = require("./Others.module");
const Dmarc_module_1 = require("./Dmarc.module");
const User_module_1 = require("./User.module");
const BlocklistMonitor_module_1 = require("./BlocklistMonitor.module");
class MailerSend {
    constructor(config) {
        this.baseUrl = "https://api.mailersend.com/v1";
        this.apiKey = config.apiKey;
        this.token = new Token_module_1.TokenModule(config.apiKey, this.baseUrl);
        this.email = new Email_module_1.EmailModule(config.apiKey, this.baseUrl);
        this.emailVerification = new EmailVerification_module_1.EmailVerificationModule(config.apiKey, this.baseUrl);
        this.sms = new SMS_module_1.SMSModule(config.apiKey, this.baseUrl);
        this.others = new Others_module_1.OthersModule(config.apiKey, this.baseUrl);
        this.dmarc = new Dmarc_module_1.DmarcModule(config.apiKey, this.baseUrl);
        this.user = new User_module_1.UserModule(config.apiKey, this.baseUrl);
        this.blocklistMonitor = new BlocklistMonitor_module_1.BlocklistMonitorModule(config.apiKey, this.baseUrl);
    }
}
exports.MailerSend = MailerSend;
