// mailersend/lib/models/Token.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenScopeType = exports.Token = void 0;
class Token {
    constructor(name, scopes, domainId) {
        this.name = name;
        this.scopes = scopes;
        this.domain_id = domainId;
    }
    setName(name) {
        this.name = name;
        return this;
    }
    setDomainId(domainId) {
        this.domain_id = domainId;
        return this;
    }
    setScopes(scopes) {
        this.scopes = scopes;
        return this;
    }
}
exports.Token = Token;
var TokenScopeType;
(function (TokenScopeType) {
    TokenScopeType["EMAIL_FULL"] = "email_full";
    TokenScopeType["DOMAINS_FULL"] = "domains_full";
    TokenScopeType["ACTIVITY_READ"] = "activity_read";
    TokenScopeType["ACTIVITY_FULL"] = "activity_full";
    TokenScopeType["ANALYTICS_READ"] = "analytics_read";
    TokenScopeType["ANALYTICS_FULL"] = "analytics_full";
    TokenScopeType["TOKENS_FULL"] = "tokens_full";
    TokenScopeType["WEBHOOKS_FULL"] = "webhooks_full";
    TokenScopeType["TEMPLATES_FULL"] = "templates_full";
    TokenScopeType["SUPPRESSIONS_READ"] = "suppressions_read";
    TokenScopeType["SUPPRESSIONS_FULL"] = "suppressions_full";
    TokenScopeType["SMS_READ"] = "sms_read";
    TokenScopeType["SMS_FULL"] = "sms_full";
    TokenScopeType["EMAIL_VERIFICATION_READ"] = "email_verification_read";
    TokenScopeType["EMAIL_VERIFICATION_FULL"] = "email_verification_full";
    TokenScopeType["INBOUNDS_FULL"] = "inbounds_full";
    TokenScopeType["RECIPIENTS_READ"] = "recipients_read";
    TokenScopeType["RECIPIENTS_FULL"] = "recipients_full";
    TokenScopeType["DOMAINS_READ"] = "domains_read";
    TokenScopeType["SENDER_IDENTITY_READ"] = "sender_identity_read";
    TokenScopeType["SENDER_IDENTITY_FULL"] = "sender_identity_full";
    TokenScopeType["USERS_READ"] = "users_read";
    TokenScopeType["USERS_FULL"] = "users_full";
    TokenScopeType["SMTP_USERS_READ"] = "smtp_users_read";
    TokenScopeType["SMTP_USERS_FULL"] = "smtp_users_full";
    TokenScopeType["DMARC_MONITORING_READ"] = "dmarc_monitoring_read";
    TokenScopeType["DMARC_MONITORING_FULL"] = "dmarc_monitoring_full";
    TokenScopeType["BLOCKLIST_MONITORING_READ"] = "blocklist_monitoring_read";
    TokenScopeType["BLOCKLIST_MONITORING_FULL"] = "blocklist_monitoring_full";
    TokenScopeType["WHATSAPP_FULL"] = "whatsapp_full";
    TokenScopeType["IFTTT"] = "ifttt";
})(TokenScopeType || (exports.TokenScopeType = TokenScopeType = {}));
