// mailersend/lib/models/User.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.UserPermission = exports.UserRole = void 0;
var UserRole;
(function (UserRole) {
    UserRole["Admin"] = "Admin";
    UserRole["Manager"] = "Manager";
    UserRole["Designer"] = "Designer";
    UserRole["Accountant"] = "Accountant";
    UserRole["CustomUser"] = "Custom User";
})(UserRole || (exports.UserRole = UserRole = {}));
var UserPermission;
(function (UserPermission) {
    UserPermission["ReadAllTemplates"] = "read-all-templates";
    UserPermission["ReadOwnTemplates"] = "read-own-templates";
    UserPermission["ManageTemplates"] = "manage-template";
    UserPermission["ReadFilemanager"] = "read-filemanager";
    UserPermission["ManageDomain"] = "manage-domain";
    UserPermission["ManageInbound"] = "manage-inbound";
    UserPermission["ManageWebhook"] = "manage-webhook";
    UserPermission["ControlSendings"] = "control-sendings";
    UserPermission["ControlTrackingOptions"] = "control-tracking-options";
    UserPermission["AccessSmtpCredentials"] = "access-smtp-credentials";
    UserPermission["ViewSmtpUsers"] = "view-smtp-users";
    UserPermission["ManageSmtpUsers"] = "manage-smtp-users";
    UserPermission["ReadRecipient"] = "read-recipient";
    UserPermission["ReadActivity"] = "read-activity";
    UserPermission["ReadEmail"] = "read-email";
    UserPermission["ReadAnalytics"] = "read-analytics";
    UserPermission["ReadSenderIdentities"] = "read-sender-identities";
    UserPermission["ManageSenderIdentities"] = "manage-sender-identities";
    UserPermission["ReadEmailVerification"] = "read-email-verification";
    UserPermission["ManageEmailVerification"] = "manage-email-verification";
    UserPermission["ManageSms"] = "manage-sms";
    UserPermission["ReadSms"] = "read-sms";
    UserPermission["ManageVerifiedRecipients"] = "manage-verified-recipients";
    UserPermission["ViewSmsWebhooks"] = "view-sms-webhooks";
    UserPermission["ManageSmsWebhooks"] = "manage-sms-webhooks";
    UserPermission["ViewSmsInbound"] = "view-sms-inbound";
    UserPermission["ManageSmsInbound"] = "manage-sms-inbound";
    UserPermission["UpdatePlan"] = "update-plan";
    UserPermission["ManageAccount"] = "manage-account";
    UserPermission["ReadInvoice"] = "read-invoice";
    UserPermission["ManageApiToken"] = "manage-api-token";
    UserPermission["ReadSuppressions"] = "read-suppressions";
    UserPermission["ManageSuppressions"] = "manage-suppressions";
    UserPermission["ReadIpAddresses"] = "read-ip-addresses";
    UserPermission["ManageIpAddresses"] = "manage-ip-addresses";
    UserPermission["ReadErrorLog"] = "read-error-log";
})(UserPermission || (exports.UserPermission = UserPermission = {}));
